querymsc.py

```diff
@@ -505,5 +507,124 @@ def query_rain_rate_one_day_1_50(self, session):
                                   History.lon,
                                   History.station_id)
                  )
- 
-        return query
\ No newline at end of file
+  
+        return query
+
+class Gumbel:
```

Comment:
Why is this a class? It seems as if an instance of this class just carries around the parameters that are provided in its constructor, those parameters being used implicitly by the other methods. Unless you will have many instances of `Gumbel` hanging around and being used to process multiple datasets, it makes more sense to me to simply pass the parameters explicitly to plain old functions, no classes required.

I'm not a huge fan of object oriented programming. It's used in far too many situations where it is not really called for, and I think this is one of those situations. 

Tell me if you think there is a good argument in favour of this class.

Reply:
A bigger issue is that this file is very long. It would probably be improved by being broken up into modules. Essentially you want to put together the final functionality by importing component functionality from various modules, each of which addresses a single concern or functionality. We can discuss.

I'm thinking that you may be trying to address this encapsulation problem by creating this (and possibly other?) class(es). Good impulse, if so, but probably wrong tool.

Reply:
If you want to hear a really long technical rant, ask me why I don't like OOP. Be forewarned.

Reply:
I wanted to address the encapsulation problem. Gumbel is not just used for rainfall_rate, but for a variety of variables, I wanted to make it as general as possible and I didn't want to make the Gumbel fitting class parameters (like return period and minimum years) a part of the WorkFlow tool... yet I still wanted them to be declared as a class parameters in a similar way that is used in many of Scikit-Learn's modules. 

Perhaps this Gumbel thing is best left out of the query-msc.py, and included as a separate class to be called by arbitrary variables? Then it begs the question whether or not Gumbel fitting belongs in query-msc or perhaps in a standalone repo.

Reply:
I love anti-OOP rants, so please invite me when you have the discussion :)

Reply:
After a bit of thought, Gumbel fitting belongs in query-msc, as a separate python file, and adding it as another tool in @pacificclimate doesn't make sense.  Especially for our rather specialized case of L-moments as estimators.